include/aspect/introspection.h

```diff
@@ -459,6 +459,20 @@ namespace aspect
       bool
       compositional_name_exists (const std::string &name) const;
 
+      /**
+       * Get the indices of the compositional fields which are of a
+       * particular type (chemical composition, porosity, etc.).
+       */
+      const std::vector<unsigned int>
+      get_indices_for_fields_of_type (const std::string &type_name) const;
```

Comment:
why is the input a string and not a value of the type enum?

Reply:
Another good point. Both changed.